react/features/settings/components/AbstractSettingsView.js

```diff
@@ -167,15 +166,15 @@ export class AbstractSettingsView extends Component<Props> {
  * @param {Object} state - The redux state.
  * @protected
  * @returns {{
- *     _profile: Object,
  *     _serverURL: string,
+ *     _settings: Object,
  *     _visible: boolean
  * }}
  */
 export function _mapStateToProps(state: Object) {
     return {
-        _profile: state['features/base/profile'],
         _serverURL: state['features/app'].app._getDefaultURL(),
+        _settings: getSettings(state),
```

Comment:
why not use `state['features/base/settings']` here?

Reply:
I wrapped it in a function to make it easier to access, I like it more this way (especially that further changes may come around settings), also replaced every occasions in the code. Don't you like it this way?

Reply:
Yes, it's more explicit. getState is useful in other places, but not here, and it should not contain more than getting the actual object from the store, otherwise we'd need getters for every piece of redux store... which doesn't sound nice.

Reply:
Actually... see previous conversation: maybe having a centralised way to retrieve redux subtrees and hide their implementation (e.g. subtree name) would be a nice thing to have. Especially as sometimes we move these things over and they change names. Isn't it better to simply change the implementation of the getSettings function when we decide to change the way we retrieve the settings, than find&replace all occurrence of the state access in the code? But maybe too complicated, not sure. Anyhow, seeing e.g. state['features/base/settings'].displayName everywhere over the web source where we need a simple string value seems to be overkill for me. Similar to how we have getParticipants or getLocalParticipant functions. But i'm open for any argument. What do you think?

Reply:
Generally, this is how we have been doing things:

* use the feature string-path-thing
* unless the function can be called with a getState function, then we add a convenience getter

Having a getter for every piece of state would just save you some typing, if you rename the feature you are still going to rename the getter. Before this PR we'd have some `getProfile` which now needs to be `getSettings`, so the improvement is negligible, IMHO.

So, the usual rule of thumb is (for me at least): "what does surrounding and alike code do?" index the state array, so please let's do that.

If you'd like to venture into making this slightly different, that's cool, but not part of this PR. I can imagine some `getFeatureState(store, featureTag)` may make some code more concise, but I'm not 100% convinced of its benefits.

Reply:
Ok, you convinced me. Does that mean then that I should remove all references of getSettings (and the function too)?

Reply:
Yes, please.